Code/Framework/AzCore/AzCore/RTTI/BehaviorContext.h

```diff
@@ -2898,7 +2942,7 @@ namespace AZ
     template<class Value>
     BehaviorDefaultValuePtr BehaviorContext::MakeDefaultValue(Value&& defaultValue)
     {
-        return !IsRemovingReflection() ? aznew BehaviorDefaultValue(AZStd::forward<Value>(defaultValue)) : nullptr;
+        return !IsRemovingReflection() ? aznew BehaviorDefaultValueTyped(AZStd::forward<Value&&>(defaultValue)) : nullptr;
```

Comment:
This is incorrect, forward should pass `Value`, not `Value&&`.
This has a slight change when supplying an rvalue reference to a function.
By default a temporary or `std::move` value deduces Value as `T` not `T&&`.
std::forward expects to be supplied a `T` or a const qualified `T&`, where it then adds an rvalue reference to the type

Reply:
Curse you C++ and your overly complex reference collapsing and type deduction rules.